Source/WTF/wtf/cf/VectorCF.h

```diff
@@ -160,6 +160,14 @@ template<typename MapLambdaType> Vector<typename LambdaTypeTraits<MapLambdaType>
     return vector;
 }
 
+inline std::span<const char> CFStringGetCStringSpan(CFStringRef string, CFStringEncoding encoding)
+{
+    auto* characters = CFStringGetCStringPtr(string, encoding);
+    if (!characters)
+        return { };
+    return unsafeMakeSpan(characters, CFStringGetLength(string));
```

Comment:
I don't think this length is right. The C string length is based on the encoding passed and won't necessarily match this length, which is the number of UTF-16 code units.

Reply:
One way to make this correct would be to name it CFStringGetASCIICStringSpan and force kCFStringEncodingASCII, like our caller used to do. (In ASCII, the number of code units is equal to the number characters.)

Reply:
At least one of the call sites was using `kCFStringEncodingISOLatin1`

Reply:
I think I can just call `strlen()`? Or even `span(const char*)` (which ends up calling strlen())

Reply:
Latin1 is still ok (1 byte per character). So you could offer an enum to pick ASCII or Latin1, or two separate functions. 

I don’t think strlen is ok because:
* if a future caller passes a multi-byte encoding, you might find a null byte before end of string
* kind of a bummer to pay the cost of enumeration when you don’t need to

Reply:
The doc for CFStringGetCStringPtr clearly states:
> The encoding must specify an 8-bit encoding.

It returns a C String, which is a null terminated string.

Reply:
I agree though that `strlen()` adds a cost. I just don't think we need to worry about multi-byte encoding since they are not supported by this API (I can add a comment to this effect for clarity).

Reply:
Anyway, to avoid the cost of strlen(), I guess your proposal is still best: Add a enum parameter to allow requesting either ASCII or Latin1 (only).

Reply:
I went with one of Geoff's proposal, separate CFStringGetASCIICStringSpan / CFStringGetLatin1CStringSpan functions and using `CFStringGetLength()`